### PR TITLE
Dev/first version

### PR DESCRIPTION
**Pull request by liubich**

### CODE REVIEW

**Comment by probil** [comment]
Doesn't work for me on Chrome/FF/Safari
It asks for location but doesn't show the weather
Screen Shot 2019-05-26 at 10 59 26

If you PR still in progress you can update add WIP status to the title, e.g. `WIP: Dev/first version`
WIP = Work In Progress

File: src/CurrentWeather/CurrentWeather.css [hunk 1]
```diff
@@ -0,0 +1,36 @@
+@import url('https://fonts.googleapis.com/css?family=Oswald:600&display=swap&subset=cyrillic');
```
Comment (probil) [inline src/CurrentWeather/CurrentWeather.css]: As I know it's better to import fonts in html rather then in css
https://stackoverflow.com/questions/12316501/including-google-web-fonts-link-or-import/12380004#12380004

**Comment by liubich** [comment]
@probil, I increased time to get location to 30 seconds. Test it, please.

File: src/CurrentWeather/CurrentWeather.css (hunk 1 above)
Comment (liubich) [reply src/CurrentWeather/CurrentWeather.css]: Interesting... Applied)

**Comment by probil** [comment]
@liubich the problem is a little bit deeper. You need to handle error & rejection in a right way
Check all possible errors here
https://www.w3schools.com/html/html5_geolocation.asp#midcontentadcontainer
> Будь ласка, надайте сторінці доступ до місцезнаходження

It's not correct. I've given access to the location

**Comment by liubich** [comment]
@probil, fixed.

**Comment by probil** [comment]
@liubich rebase on latest master to make Circle CI verify this PR

**Comment by liubich** [comment]
@probil, done

File: src/redux/reducers/reducer.js [hunk 2]
```diff
@@ -0,0 +1,14 @@
+export default function educer(state, action) {
```
[the remark this answers is not in the document]
Comment (liubich) [reply src/redux/reducers/reducer.js]: Of course, it should be `reducer`

File: src/redux/actions/actions.js [hunk 3]
```diff
@@ -0,0 +1,116 @@
+/* eslint-disable no-console */
```
Comment (probil) [inline src/redux/actions/actions.js]: this line is not needed

File: src/redux/actions/actions.js [hunk 4]
```diff
@@ -0,0 +1,116 @@
...
+    description: json.weather[0].description,
+    dateTime: getFormattedDateStr(json.dt),
+    windSpeed: parseInt(json.wind.speed),
+    windDirection: getWindDirection(parseInt(json.wind.deg)),
+    pressure: parseInt(json.main.pressure)
+  };
+};
+
+const getFormattedDateStr = UNIXdate => {
+  const reqDate = new Date(Date(parseInt(UNIXdate)));
```
[the remark this answers is not in the document]
Comment (liubich) [reply src/redux/actions/actions.js]: Right, it was needed for a single string expression and I forgot to remove it after refactoring.

File: src/CurrentWeather/CurrentWeather.js [hunk 5]
```diff
@@ -0,0 +1,47 @@
+import React from "react";
+import { connect } from "react-redux";
+import PropTypes from "prop-types";
+import "./CurrentWeather.css";
+
+const CurrentWeather = props => {
+  return (
+    <div className="current-weather">
+      <h2 className="current-weather__header">
+        Погода у {props.currentWeather.place},{" "}
```
[the remark this answers is not in the document]
Comment (liubich) [reply src/CurrentWeather/CurrentWeather.js]: It was added by Prettier

[the remark this answers is not in the document]
Comment (probil) [reply src/CurrentWeather/CurrentWeather.js]: I have `prettier` sometimes

[the remark this answers is not in the document]
Comment (probil) [reply src/CurrentWeather/CurrentWeather.js]: Can you clean it up?

[the remark this answers is not in the document]
Comment (liubich) [reply src/CurrentWeather/CurrentWeather.js]: Yes, I did

File: src/redux/actions/actions.js [hunk 6]
```diff
@@ -0,0 +1,117 @@
...
+    [error.PERMISSION_DENIED]:
+      "Будь ласка, надайте сторінці доступ до місцезнаходження",
+    [error.POSITION_UNAVAILABLE]:
+      "Недоступні дані про поточне місцезнаходження",
+    [error.TIMEOUT]: "Вийшов час для визначення місцезнаходження"
+  };
+  return errorCodeToDescription[error.code] || "Невідома помилка";
+};
+
+const getCurrentWeatherAPIUrl = (latitude, longitude, APIkey) => {
```
Comment (probil) [inline src/redux/actions/actions.js]: looks good 👍 
We can make it even better by doing so
```js
const getCurrentWeatherAPIUrl = ({ latitude, longitude, APIkey, lang = 'ua', units='metric' }) => {
  return `https://api.openweathermap.org/data/2.5/weather?lat=${latitude}&lon=${longitude}&APPID=${APIkey}&lang=${lang}&units=${units}`;
};
```

Comment (probil) [reply src/redux/actions/actions.js]: It's better to pass data as object then just a list:
```js
const getCurrentWeatherAPIUrl = () => {}
getCurrentWeatherAPIUrl(45.234,54.3434,'dsfdfsdf'); // it's not clear which is `lat` and which is `lon`
getCurrentWeatherAPIUrl({ lat: 45.234, lan: 54.3434, APIKey: 'dsfdfsdf' }); // much clear
```
https://refactoring.guru/uk/smells/long-parameter-list

File: src/CurrentWeather/CurrentWeather.js [hunk 7]
```diff
@@ -28,18 +28,21 @@ const CurrentWeather = props => {
...
             >
-              &#8681;{" "}
-            </span>
-            {props.currentWeather.windDirection},{" "}
-            {props.currentWeather.windSpeed} м/с
-          </p>
+              &#8681;
+            </div>
+            <div className="current-weather__wind-caption">
+              {props.currentWeather.windDirection},{" "}
```
Comment (probil) [inline src/CurrentWeather/CurrentWeather.js]: remove `{" "}` please

**Pull request merged**